packages/ssz/src/type/stableContainer.ts

```diff
@@ -0,0 +1,794 @@
+      this.jsonKeyToFieldName[jsonKey] = fieldName;
+    }
+
+    const {minLen, maxLen, fixedSize} = precomputeSizes(this.fieldsEntries);
+    this.minSize = minLen;
+    this.maxSize = maxLen;
+    this.fixedSize = fixedSize;
+
+    // TODO: This options are necessary for ContainerNodeStruct to override this.
+    // Refactor this constructor to allow customization without pollutin the options
+    this.TreeView = opts?.getContainerTreeViewClass?.(this) ?? getContainerTreeViewClass(this);
+    this.TreeViewDU = opts?.getContainerTreeViewDUClass?.(this) ?? getContainerTreeViewDUClass(this);
+  }
+
+  static named<Fields extends Record<string, Type<unknown>>>(
+    fields: Fields,
+    maxFields: number,
+    opts: Require<StableContainerOptions<Fields>, "typeName">
+  ): StableContainerType<Fields> {
+    return new (namedClass(StableContainerType, opts.typeName))(fields, maxFields, opts);
+  }
+
+  defaultValue(): ValueOfFields<Fields> {
+    const value = {} as ValueOfFields<Fields>;
+    for (const {fieldName, fieldType, optional} of this.fieldsEntries) {
+      value[fieldName] = (optional ? null : fieldType.defaultValue()) as ValueOf<Fields[keyof Fields]>;
+    }
+    return value;
+  }
+
+  getView(tree: Tree): ContainerTreeViewType<Fields> {
+    return new this.TreeView(this, tree);
+  }
+
+  getViewDU(node: Node, cache?: unknown): ContainerTreeViewDUType<Fields> {
+    return new this.TreeViewDU(this, node, cache);
+  }
+
+  cacheOfViewDU(view: ContainerTreeViewDUType<Fields>): unknown {
+    return view.cache;
+  }
+
+  commitView(view: ContainerTreeViewType<Fields>): Node {
+    return view.node;
+  }
+
+  commitViewDU(view: ContainerTreeViewDUType<Fields>): Node {
+    view.commit();
+    return view.node;
+  }
+
+  // Serialization + deserialization
+  // -------------------------------
+  // Containers can mix fixed length and variable length data.
+  //
+  // Fixed part                         Variable part
+  // [field1 offset][field2 data       ][field1 data               ]
+  // [0x000000c]    [0xaabbaabbaabbaabb][0xffffffffffffffffffffffff]
+
+  value_serializedSize(value: ValueOfFields<Fields>): number {
+    let totalSize = Math.ceil(this.fieldsEntries.length / 8);
+    for (let i = 0; i < this.fieldsEntries.length; i++) {
+      const {fieldName, fieldType, optional} = this.fieldsEntries[i];
+      // skip optional fields with nullish values
+      if (optional && value[fieldName] == null) {
+        continue;
+      }
+
+      // Offset (4 bytes) + size
+      totalSize +=
+        fieldType.fixedSize === null ? 4 + fieldType.value_serializedSize(value[fieldName]) : fieldType.fixedSize;
+    }
+
+    return totalSize;
+  }
+
+  value_serializeToBytes(output: ByteViews, offset: number, value: ValueOfFields<Fields>): number {
+    // compute active field bitvector
+    const activeFields = BitArray.fromBoolArray(this.fieldsEntries.map(({fieldName}) => value[fieldName] != null));
```

Comment:
need to pad to make it a `BitVector`? from the spec: "The Bitvector[N] is padded with False bits up through length N"

Reply:
`BitArray` is only the value not type. Still follow BitVector spec there